src/renderer/js/redux/reducers/app.js

```diff
@@ -1,22 +1,45 @@
-import * as types from "constants/action_types";
-import * as modalTypes from "constants/modal_types";
+// @flow
 
+import * as types from "../../constants/action_types";
+import * as modalTypes from "../../constants/modal_types";
+
+// $FlowIssue: Flow cannot find election
 const { remote } = require("electron");
+
 const application = remote.app;
 const win = remote.BrowserWindow.getFocusedWindow();
 
 const reducers = {};
-const defaultState = {
+type appState = {
+  isLoaded: boolean,
+  modal: mixed,
+  modalProps: mixed,
+  platform: string,
+  upgradeSkipped: boolean,
+  daemonVersionMatched: ?boolean,
+  daemonReady: boolean,
+  hasSignature: boolean,
+  badgeNumber: number,
+  volume: number,
+  downloadProgress?: number,
+  upgradeDownloading?: boolean,
+  upgradeDownloadComplete?: boolean,
+  checkUpgradeTimer?: mixed,
+  isUpgradeAvailable?: boolean,
+  isUpgradeSkipped?: boolean,
+  snackBar?: mixed,
+};
+const defaultState: appState = {
   isLoaded: false,
   modal: null,
   modalProps: {},
   platform: process.platform,
-  upgradeSkipped: sessionStorage.getItem("upgradeSkipped"),
+  upgradeSkipped: sessionStorage.getItem("upgradeSkipped") === "true",
```

Comment:
Can you clarify this change?

(And btw, we want to be killing this storage pattern)

Reply:
`sessionStorage.getItem("upgradeSkipped")` returns the string: `"true"` so this is, essentailly, casting it to a boolean.

Reply:
@liamcardenas but this changeset also changes this value from `true` to `"true"`.

(If you just want to cast, I'd propose either a typical cast of `Boolean(foo)` or `!!foo`)

Reply:
I think he's saying no matter how it was stored in `sessionStorage` it was still being returned as `"true"`. 

Using the `!!` operator or `Boolean` will still return `true` for `"false"` strings because strings are truthy.

@liamcardenas is that right?

Reply:
that doesn't work, because `Boolean('false')` and `!!'false'` evaluate to `true`

This is the best way to get a true value that is recorded as a string. The alternative would be `JSON.parse(foo)`, but I wouldn't want to be able to introduce arbitrary, non-boolean values.

Reply:
Wow, sure enough, the Storage API changes the type! 
```
window.sessionStorage.setItem("foo", true);
undefined
window.sessionStorage.getItem("foo");
"true"
```

Absolute insanity!

Reply:
I know, I wouldn't have caught this if flow didn't tell give me a type error for trying to input/expect booleans where it is supposed to be a string.

That's the beauty of type checking, this could've led to some pretty nasty bugs.